### PR TITLE
Check if the format of override model config has extra fields

### PR DESCRIPTION
**Pull request by krishung5**
Loading model with override model config is working properly. The [issue](https://github.com/triton-inference-server/server/issues/4416) was that the config that the user used has some extra fields but no error message was shown.
Fix the error checking for invalid override config in Core repo: https://github.com/triton-inference-server/core/pull/124
Client repo: https://github.com/triton-inference-server/client/pull/170

### CODE REVIEW

File: src/grpc_server.cc [hunk 1]
```diff
@@ -2037,6 +2037,17 @@ CommonHandler::SetUpAllRequests()
           std::vector<const TRITONSERVER_Parameter*> const_params;
           for (const auto& param_proto : request.parameters()) {
             if (param_proto.first == "config") {
+              if ((param_proto.second.string_param().find("config") !=
+                   std::string::npos) ||
+                  (param_proto.second.string_param().find("parameters")) !=
+                      std::string::npos) {
```
Comment (GuanLuo) [inline src/grpc_server.cc:2043]: I don't think this is the proper way to check it, what if the user nests "some_other_field"? This check is going to miss that. The question seems to be why the model manager fails to recognize the config is ill-formed and reject the model loading.

Comment (krishung5) [reply src/grpc_server.cc]: Seems like I didn't think it thoroughly enough. I have fixed the error checking for the ill-formed config in the core repo. Reverted the previous changes. Please review.

**Review by dyastremsky — DISMISSED** [review]
Looks good! Great work on these tickets.

**Comment by krishung5** [comment]
Updated test scripts according to new changes.

**Review by dyastremsky — DISMISSED** [review]

**Comment by krishung5** [comment]
@dyastremsky There was some typo, my bad. Please re-approve, thank you!
Passed on the [CI](https://gitlab-master.nvidia.com/dl/dgx/tritonserver/-/pipelines/6114332).

**Review by dyastremsky — APPROVED** [review]

**Pull request merged**